eng/scripts/vcpkg.ps1

```diff
@@ -0,0 +1,52 @@
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $Ref = 'master',
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Dependencies,
+
+    [Parameter()]
+    [string] $TargetPath ="$env:TEMP/$([guid]::NewGuid())",
+
+    [Parameter()]
+    [switch] $DevOpsLogging = ($null -ne $env:SYSTEM_TEAMPROJECTID)
+)
+
+function SetEnvironmentVariable {
+    param(
+        [string] $Name,
+        [string] $Value
+    )
+
+    if ($devOpsLogging) {
+        Write-Host "##vso[task.setvariable variable=$Name]$($Value)"
+    } else {
+        Write-Verbose "Setting local environment variable: $Name = ***"
+        Set-Item -Path "env:$Name" -Value $Value
+    }
+}
+
+try {
+    New-Item -ItemType Directory -Path $TargetPath -Force
+    Push-Location $TargetPath
+
+    git clone https://github.com/Microsoft/vcpkg .
+    git checkout (Get-Content "$PSScriptRoot/../vcpkg-ref.txt")
+
+    if ($IsWindows) {
+        .\bootstrap-vcpkg.bat
+        .\vcpkg.exe install $Dependencies.Split(' ')
```

Comment:
Will the `.Split` fail if `$Dependencies` is empty? Maybe we should skip the installation if we don't have any dependencies.

Reply:
We do skip installation if the dependencies are empty: 

```
        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
```